staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/compile.go

```diff
@@ -198,19 +199,19 @@ func (c compiler) CompileCELExpression(expressionAccessor ExpressionAccessor, op
 			reason = fmt.Sprintf("must evaluate to one of %v", returnTypes)
 		}
 
-		return resultError(reason, apiservercel.ErrorTypeInvalid)
+		return resultError(reason, apiservercel.ErrorTypeInvalid, nil)
 	}
 
 	_, err = cel.AstToCheckedExpr(ast)
 	if err != nil {
 		// should be impossible since env.Compile returned no issues
-		return resultError("unexpected compilation error: "+err.Error(), apiservercel.ErrorTypeInternal)
+		return resultError("unexpected compilation error: "+err.Error(), apiservercel.ErrorTypeInternal, nil)
```

Comment:
Since we are passing nil here, we should ensure that we check for this value to be nil wherever its used, I couldn't find any instances of usage of this field in this PR but was just something that stood out for me. Maybe replacing nil here with a "apiservercel.UnknownError" or something would be better? Leave it to you.

Reply:
This is the code we would never expect to hit :) So we don't have a specific err type for this :)